autogpt_platform/backend/backend/copilot/bot/adapters/slack/history.py

```
@@ -0,0 +1,112 @@
+"""Thread history for the Slack adapter.
+
+A first @-mention into a thread the bot doesn't own pulls the thread's recent
+messages into the prompt. Slack's ``conversations.replies`` pages oldest-first
+from the parent, so the fetch follows the cursor to the end and keeps only the
+tail, which the shared char budget then trims newest-first.
+"""
+
+import asyncio
+import logging
+from collections import deque
+from typing import Any, Awaitable, Callable, Optional
```

Ruff (0.16.1)

[warning] Import from `collections.abc` instead: `Awaitable`, `Callable`

Import from `collections.abc`

(UP035)